automaxlair/MaxLairInstance.py

```diff
@@ -410,9 +410,9 @@ def calculate_ore_cost(self, num_resets: int) -> int:
         """Calculate the prospective Dynite Ore cost of resetting the game."""
         return 0 if num_resets < 3 else min(10, num_resets)
 
-    def check_sufficient_ore(self, num_resets: int) -> bool:
+    def check_sufficient_ore(self, needed_ore: int) -> bool:

```

Comment:
I see that there was an error with the original logic, as you would have to pay twice for two resets. However I'm not sure this approach is the cleanest. What if it took the number of additional resets as the argument? The  you could sum up all the ore costs by internally calling calculate_ore_cost. LMK if that doesn't make sense and I can take a crack at it.

Reply:
Yes it makes sense, let me do that if you prefer !